resources/design-document.md

```diff
@@ -0,0 +1,206 @@
+
+U9. _As a [MC player], I want to view my w/l stats by the different aspect I played_
+
+
+## 3.1 Stretch Use cases:
+
+SU1. _As a [MC player], I want to view a list of my favorite heroes_
+
+SU2. _As a [MC player], I want to query all my games in the game log by hero, aspect, villain, etc_
+
+
+
+# 4. Project Scope
+
+### 4.1. In Scope
+
+- Creating, retrieving, updating, and deleting a game log.
+- Viewing all your game logs.
+- Viewing your overall w/l stats.
+- Viewing your w/l stats with a certain hero or villain.
+- Viewing your w/l stats with a certain aspect you played.
+
+
+### 4.2. Out of Scope
+
+- The Stretch goals.
+- Being able to purchase MC cards from our app.
+- Organizing/keeping up with your MC collection.
+
+# 5. Proposed Architecture Overview
+
+This initial iteration will provide the minimum lovable product (MLP) including creating, retrieving, updating,
+and deleting a game log for game log requests. It will also allow users to view their overall w/l stats and their w/l
+stats with certain heroes, villains and aspect.
+
+We will use API Gateway and AWS Lambda to create the following endpoints:
+- CreateGameLog
+- GetGameLog
+- GetAllGameLogs
+- UpdateGameLog
+- DeleteGameLog
+- GetOverallStats
+- GetHeroStats
+- GetVillainStats
+- GetAspectStats
+
+
+I will store GameLogs and heroes and villains in separate DynamoDB tables.
+
+
+# 6. API
+
+## 6.1. Public Models
+
+```
+// GameLogModel
+ 
+email // string 
+gameId // string
+date // string (converted date)
+outcome // enum
+aspect // enum
+gameCharacters // List<PlayerCharacter>
+
+```
+
+```
+
+// PlayerCharacterModel
+
+name // string
+roles // List<Enum>   
+
+```
+
+
+## 6.2 Endpoints
+
+### 6.2.0 _Create GameLog Endpoint_
+- Accepts `POST` requests to `/gamelogs`
+- Accepts data to create a new gameLog with a provided gameId, date, w/l outcome, aspect, hero,
+  and villain. Returns the new GameLog, including a game ID assigned by the service.
+
+### 6.2.1 _Get single GameLog Endpoint_
+- Accepts `GET` requests to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts an email and game ID and returns the corresponding GameLogModel.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.1.1 _Get all GameLogs by player Endpoint_
+- Accepts `GET` requests to `/gamelogs/:`
+- Takes the email from cognito
+- Accepts an email and returns the corresponding list of GameLogModels.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.2 _Delete GameLog Endpoint_
+- Accepts a `DELETE` request to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts a game ID and returns the corresponding Deleted GameLogModel.
+    - If the given game ID is not found, will throw a `GameNotFoundException`
+
+### 6.2.3 _Update GameLog Endpoint_
+- Accepts a `PUT` request to `/gamelogs/:gameId`
+- Takes the email from cognito
+- Accepts data to update a gameLog and returns the updated gameLog.
```

Comment:
Similar to the note above on the user story - what data will you be able to update?